ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/ProjectImporter.java

```diff
@@ -230,21 +219,36 @@ public void onFailure(Throwable caught) {
                   @Override
                   public void onSuccess(OAuthStatus result) {
                     if (!result.equals(OAuthStatus.NOT_PERFORMED)) {
-                      doImport(path, sourceStorage)
-                          .then(
-                              new Operation<Project>() {
-                                @Override
-                                public void apply(Project project) throws OperationException {
-                                  callback.onSuccess(project);
-                                }
-                              })
-                          .catchError(
-                              new Operation<PromiseError>() {
-                                @Override
-                                public void apply(PromiseError error) throws OperationException {
-                                  callback.onFailure(error.getCause());
-                                }
-                              });
+                      oAuthServiceClient.getToken(
+                          providerName,
+                          new AsyncRequestCallback<OAuthToken>(
+                              unmarshallerFactory.newUnmarshaller(OAuthToken.class)) {
+                            @Override
+                            protected void onSuccess(OAuthToken result) {
+                              final SourceStorageImpl copySourceStorage =
```

Comment:
please, can you explain why do you need a copy here? Are you not able to put required parameters in existing source storage like you do here https://github.com/eclipse/che/pull/7767/files#diff-cf793e40fa0862248ddeaa02affbd100R177 ?

Reply:
overengeneering :)